commands/dht.go

```diff
@@ -0,0 +1,163 @@
+package commands
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"time"
+
+	cmds "gx/ipfs/QmQtQrtNioesAWtrx8csBvfY37gTe94d6wQ3VikZUjxD39/go-ipfs-cmds"
+	"gx/ipfs/QmR8BauakNcBa3RbE4nbQu76PDiJgoQgz8AJdhJuiU4TAw/go-cid"
+	pstore "gx/ipfs/QmRhFARzTHcFh8wUxwN5KvyTGq73FLC65EfFAhz8Ng7aGb/go-libp2p-peerstore"
+	notif "gx/ipfs/QmWaDSNoSdSXU9b6udyaq9T8y6LkzMwqWxECznFqvtcTsk/go-libp2p-routing/notifications"
+	cmdkit "gx/ipfs/Qmde5VP1qUkyQXKCfmEUA7bP64V2HAptbJ7phuPp7jXWwg/go-ipfs-cmdkit"
+)
+
+const (
+	dhtVerboseOptionName   = "verbose"
+	numProvidersOptionName = "num-providers"
+)
+
+// Note, most of this is copied directly from go-ipfs (https://github.com/ipfs/go-ipfs/blob/master/core/commands/dht.go).
+// A few simple modifications have been adapted for filecoin.
```

Comment:
This copy-pasting is a lot of duplication across projects.  However my guess is that this is the right way to do it because we've done the same thing with other commands in the past and neither project likely wants to deal with abstractions that generalize across core APIs of the other project.  Please let me know if I should be abstracting this logic somewhere else to share amongst projects instead.